dist/tools/compile_test/compile_test.py

```diff
@@ -86,12 +89,33 @@ def get_results_and_output_from(fd):
             output.write(line)
 
 
+def get_app_dirs():
+    yield from check_output(["make", "-f", "makefiles/app_dirs.inc.mk", "info-apps"]) \
+            .decode("utf-8", errors="ignore")\
+            .split()
+
+
+def split_apps_by_dir(app_dirs):
+    """ creates a dictionary as follows:
+    { "examples": ["hello_world", "gnrc_networking" ],
+      "tests": ["minimal", "fmt_print" ]
+      }
+    """
+    res = defaultdict(list)
+    for app_dir in app_dirs:
+        folder, app = app_dir.split("/", 1)
+        res[folder].append(app)
+
+    return res
```

Comment:
I think it is better to return a normal dict `return dict(res)` as using a default dict without knowing it is one can lead to trouble.

Reply:
I do not find any reference about it tough.

Reply:
then let's keep it.